index_configuration_tool/tests/test_main.py

```diff
@@ -0,0 +1,172 @@
+    def test_get_endpoint_info(self):
+        # Simple base case
+        test_config = create_plugin_config(["test"])
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL enabled
+        test_config = create_plugin_config(["test"], True)
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("https://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL disabled, invalid auth config
+        test_config = create_plugin_config(["test"], False, "test")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertFalse(result[1])
+        # SSL disabled, valid auth config
+        test_config = create_plugin_config(["test"], user="user", password="password")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("http://test/", result[0])
+        self.assertTrue(result[1])
+        # Array of hosts uses the first entry
+        test_config = create_plugin_config(["test1", "test2"], True, "user", "password")
+        result = main.get_endpoint_info(test_config)
+        self.assertEqual("https://test1/", result[0])
+        self.assertTrue(result[1])
+
+    def test_print_report(self):
+        # Base case should return an empty list
+        self.assertFalse(main.print_report(dict(), dict()))
+        base_data = {
+            "index1": {
+                "settings": {
+                    "bool_key": True,
+                    "num_key": 1
+                },
+                "mappings": {
+                    "str_key": "value"
+                }
+            },
+            "index2": {}
+        }
+        result = main.print_report(base_data, dict())
+        self.assertEqual(2, len(result))
+        self.assertTrue("index1" in result)
+        self.assertTrue("index2" in result)
+        # Index present with exact match
+        test_data = copy.deepcopy(base_data)
+        del test_data["index2"]
+        self.assertFalse(main.print_report(test_data, test_data))
+        # Index has conflict in settings
+        test_data["index1"]["settings"]["num_key"] = -1
+        result = main.print_report(base_data, test_data)
+        self.assertEqual(1, len(result))
+        self.assertTrue("index2" in result)
+        # Index has conflict in mappings
+        test_data = copy.deepcopy(base_data)
+        del test_data["index1"]["mappings"]["str_key"]
+        self.assertFalse(main.print_report(base_data, test_data))
+
+    def test_validate_plugin_config_unsupported_endpoints(self):
+        with self.assertRaises(ValueError):
+            # No supported endpoints
+            main.validate_plugin_config(BASE_DATA, TEST_KEY)
+
+    def test_validate_plugin_config_missing_host(self):
+        with self.assertRaises(ValueError):
+            # Missing hosts value
+            test_data = create_config_section({})
```

Comment:
Minor thing for these next few tests, I would pull the `test_data = ...` line _out_ of the `with` block, so the `assertRaises` is very clear about what part you expect to raise errors.